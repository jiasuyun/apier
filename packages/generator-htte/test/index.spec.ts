import Generator from "../src";
import { factory, toYaml } from "@jiasuyun/apier-test-utils-generator";

const gen = name => toYaml(factory(api => new Generator(api).value)(name));

test("generate", () => {
  expect(gen("general")).toMatchSnapshot();
});
