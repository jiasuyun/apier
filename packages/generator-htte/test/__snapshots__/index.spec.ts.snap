// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate 1`] = `
"- test:
    describe: 获取模型
    units:
      - describe: 获取模型-成功
        include: getModel
        req:
          headers:
            X-ORG-ID: 32432
          params:
            id: 32234
          query:
            pageSize: 10
            pageNo: 1
          body:
            integer: 32
            number: 16.23
            bool: true
            'null': null
            array:
              - '1'
              - '2'
            object:
              foo: 3
        res:
          status: 200
          body:
            integer: 32
            number: 16.23
            bool: true
            'null': null
            array:
              - foo: 1
              - bar: 2
            object:
              password: a234324
      - describe: 获取模型-失败
        include: getModel
        metadata:
          skip: true
  define:
    getModel:
      method: get
      url: '/model/{id}'
"
`;
